.github/workflows/main.yml

```diff
@@ -145,8 +145,21 @@ jobs:
         run: |
           python scripts/make_docs.py
           git diff --exit-code
+      - name: Upload the example installers as artifacts
+        if: github.event_name == 'pull_request' && matrix.python-version == '3.9'
+        uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
+        with:
+          name: installers-${{ runner.os }}-${{ github.sha }}-${{ github.run_id }}-${{ github.run_number }}-${{ github.run_attempt }}
+          path: "${{ runner.temp }}/examples_artifacts"
+          retention-days: 7
+
+  report:
+    name: Report failures
+    needs: [tests]
+    if: always() && github.event_name == 'push' && github.ref_name == 'main' && needs.tests.result == 'failure'
+    runs-on: ubuntu-latest
+    steps:
       - name: Report failures
```

Comment:
I think we need to checkout the repo before this action so it can access the template MD.

Reply:
```suggestion
      - name: Retrieve the source code
        uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
      - name: Report failures
```